kafka/adminapi_test.go

```diff
@@ -918,6 +918,80 @@ func testAdminAPIsDeleteRecords(what string, a *AdminClient, expDuration time.Du
 	}
 }
 
+func testAdminAPIsElectLeaders(what string, a *AdminClient, expDuration time.Duration, t *testing.T) {
+	topic := "test"
+	partition := int32(0)
+	ctx, cancel := context.WithTimeout(context.Background(), expDuration)
+	defer cancel()
+	topicPartition := []TopicPartition{{Topic: &topic, Partition: partition}}
+	emptyTopicPartition := []TopicPartition{}
+	electLeaderRequestPreferred := ElectLeadersRequest{
+		electionType: ElectionTypePreferred,
+		partitions:   topicPartition,
+	}
+	electLeadersRequestUnclean := ElectLeadersRequest{
+		electionType: ElectionTypeUnclean,
+		partitions:   topicPartition,
+	}
+
+	_, err := a.ElectLeaders(ctx, electLeaderRequestPreferred, SetAdminRequestTimeout(time.Second))
+	if err == nil || ctx.Err() != context.DeadlineExceeded {
+		t.Fatalf("Expected context deadline exceeded, got %s and %s\n",
+			err, ctx.Err())
+	}
+
+	_, err = a.ElectLeaders(ctx, electLeadersRequestUnclean, SetAdminRequestTimeout(time.Second))
+	if err == nil || ctx.Err() != context.DeadlineExceeded {
+		t.Fatalf("Expected context deadline exceeded, got %s and %s\n",
+			err, ctx.Err())
+	}
+
+	// Invalid option value
+	_, err = a.ElectLeaders(ctx, electLeaderRequestPreferred, SetAdminRequestTimeout(-1))
+	if err == nil || err.(Error).Code() != ErrInvalidArg {
+		t.Fatalf("Expected ErrInvalidArg, not %v", err)
+	}
+
+	_, err = a.ElectLeaders(ctx, electLeadersRequestUnclean, SetAdminRequestTimeout(-1))
+	if err == nil || err.(Error).Code() != ErrInvalidArg {
+		t.Fatalf("Expected ErrInvalidArg, not %v", err)
+	}
+
+	for _, options := range [][]ElectLeadersAdminOption{
+		{},
+		{SetAdminRequestTimeout(time.Second)},
+	} {
+		// empty list should fail
```

Comment:
Change this after the librdkafka change, so the empty list fails, but a `nil` list times out instead

Reply:
Actually after the librdkafka change, both empty list and nil partitions is valid.
If there is an empty list in input, we are getting an empty list in the output and for nil, all partitions.